### ISSUE TITLE
[Question] Where to put certificate

### ISSUE THREAD

**Issue by palasinio** [arendruni/homebridge-pihole#28]
Hi, I use a self signed certificate pair to use SSL.

Your Plugin reports then:
```
Error: unable to verify the first certificate
    at TLSSocket.onConnectSecure (_tls_wrap.js:1502:34)
    at TLSSocket.emit (events.js:314:20)
    at TLSSocket._finishInit (_tls_wrap.js:937:8)
    at TLSWrap.ssl.onhandshakedone (_tls_wrap.js:711:12) {
  code: 'UNABLE_TO_VERIFY_LEAF_SIGNATURE',
  config: {
    url: 'api.php',
    method: 'get',
    params: { status: 1 },
    headers: {
      Accept: 'application/json, text/plain, */*',
      'User-Agent': 'axios/0.19.2'
    },
    baseURL: 'https://pi.hole:443/admin/',
    transformRequest: [ [Function: transformRequest] ],
    transformResponse: [ [Function: transformResponse] ],
    timeout: 0,
    adapter: [Function: httpAdapter],
    responseType: 'json',
    xsrfCookieName: 'XSRF-TOKEN',
    xsrfHeaderName: 'X-XSRF-TOKEN',
    maxContentLength: -1,
    validateStatus: [Function: validateStatus],
    data: undefined
  },
```
Where can I place the certificate to get it work again?

Regards,

**Comment by arendruni** [comment]
Hi @palasinio, 
Sorry for the late answer, I can’t help you so much since I have never configured the plugin to use https, I have in plan to switch my personal configuration to use it, but I don’t know when I will.
Did you have a look at #9, in which a user was having a [problem ](https://github.com/arendruni/homebridge-pihole/issues/9#issuecomment-642656041)configuring the certificate within node?

**Comment by rmhall** [comment]
Just installed Hoobs on a Raspberry Pi 4 along side Apache, Nginx etc, and then installed Pihole, and am using a self-signed certificate and ran across this in terms of getting errors with this plugin to connect properly - errors around the self-signed certificate.,

A quick and dirty fix for this is to edit index.ts or the dist of index.js and add this line after "use strict"

process.env.NODE_TLS_REJECT_UNAUTHORIZED = '0';

This will allow self-signed certs to work. Technically this should be set through a property in the Axios requests more info here: https://github.com/axios/axios/issues/535

There ideally should be a config setting in this plugin to allow a user to indicate they are using a self-signed cert, which would then change the properties of the axis HTTPS agent accordingly.

**Comment by arendruni** [comment]
As @rmhall suggested, I added a new config setting in the plugin which changes the properties of Axios:

`rejectUnauthorized`: If the HTTPS agent should check the validity of SSL cert, set it to false to allow self-signed certs to work. Default is true.

I released a new version (v0.3.2) that integrates the new config setting.

**Comment by palasinio** [comment]
Big THX @arendruni . In my case your change works as intended.
